pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionFilePackage.java

```diff
@@ -79,7 +79,7 @@ public FunctionFilePackage(File file, String narExtractionDirectory, boolean ena
                         }
                         classFileLocators.add(locator);
                     } catch (IOException e) {
-                        throw new UncheckedIOException(e);
+                        throw new UncheckedIOException(new IOException("Loading '" + classpath + "' failed due to " + e.getMessage(), e));
```

Comment:
```suggestion
                        throw new UncheckedIOException("Loading '" + classpath + "' failed due to " + e.getMessage(), e);
```

There's no need to wrap the exception in yet another nested IOException.